### PR TITLE
Update image reference to quay.io/konflux-ci

### PR DESCRIPTION
**Pull request by chmeliik**
[STONEBLD-2336](https://issues.redhat.com//browse/STONEBLD-2336)

The downstream release configuration has been updated to release the
repo to quay.io/konflux-ci. Update the kustomize config accordingly.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

### CODE REVIEW

**Comment by chmeliik** [comment]
So, the e2e tests will never pass.

When deploying Konflux, the e2e-tests repo clones infra-deployments and does some magic, the result of which is https://github.com/redhat-appstudio-qe/infra-deployments/commit/ac24a623a4dc1e9f820db7ec900d212cf34bdf39

[This](https://github.com/redhat-appstudio-qe/infra-deployments/commit/ac24a623a4dc1e9f820db7ec900d212cf34bdf39#diff-0ffefc714cb83d66d4c09faf1cb6bc8c143e2cb5674741c6ca8f0edd34a571d2R8-R10) is the part that normally replaces the placeholder `quay.io/redhat-appstudio/build-service:next` image with the new one built by Prow. But with the change in this PR, the name no longer matches - `quay.io/redhat-appstudio/build-service` in infra-deployments vs. `quay.io/konflux-ci/build-service` here.

In order to make the names match again, we have to update infra-deployments. But we can't update infra-deployments _first_ - the name also wouldn't match, and that would cause even bigger issues than just failing e2e tests.

TLDR, we need to force-merge this and _then_ make the names match by fixing infra-deployments

_I :heart: kustomize_
